Model/Sync/Category/Processor/ProcessByCategory.php

```diff
@@ -105,7 +105,29 @@ public function process($retryCategories = [], $storeIds = [])
                         $this->config->getStoreName($storeId) . PHP_EOL;
                 }
                 $this->emulateFrontendArea($storeId);
+                $syncShouldProgress = true;
```

Comment:
No need to add this flag, just stop the environment emulation and continue from within the condition handling.

Reply:
Okay, we will make these changes